protocol/syncer.go

```diff
@@ -144,7 +144,8 @@ type Syncer struct {
 	logger     hclog.Logger
 	blockchain blockchainShim
 
-	peers map[peer.ID]*syncPeer // TODO: Remove
+	peers     map[peer.ID]*syncPeer
+	peersLock sync.RWMutex
```

Comment:
I think we can improve performance here even more than with using a RW mutex.

Since our peers are inside of a map, we currently have only 1 mutex that, when you grab it, you hold the access to the entire map.
This means that if I'm accessing the `syncPeer` at key value `peerID1` (not actual peer.ID), I'll be holding up any process that wants to access the `syncPeer` at key value `peerID2`.

I have two suggestions:
1. Create a separate mux map, where you would map the peer.ID -> mux. This way, you hold the mux to a single entry, not the entire map
2. Use `sync.Map` to accomplish the behavior mentioned in the first suggestion. [Godoc](https://pkg.go.dev/sync#Map). This is definitely the better route, as this solution takes into account any optimizations that we might miss.